pytorch_lightning/strategies/strategy.py

```diff
@@ -442,7 +445,12 @@ def teardown(self) -> None:
         It is the right place to release memory and free other resources.
         """
         optimizers_to_device(self.optimizers, torch.device("cpu"))
+
+        if self.lightning_module is not None:
+            log.detail(f"{self.__class__.__name__}: moving model to CPU")
+            self.lightning_module.cpu()
```

Comment:
Why not have this be part of the GPUAccelerator teardown?

This might be limiting for other accelerators (but that's an issue in current master too)

Reply:
AFAIK our design is to move models back to cpu in general after we are done, no? When we are supposed to release all device memory claimed, there is no other way then to move the model to cpu imo. And this should be handled by the strategy and not the accelerator since the strategy is owning the model :)